atlp-rwanda/script-heroes-bn-backend
# #174797543 CRUD on Bookings

### What does this PR do?

1. Allow users to book an accomodation
2. Allow users to get all bookings
3. Allow users to get a single booking by id
4. Allow users to update a booking using a id
5. Allow users to delete a booking using its id
6. Allow users to delete all bookings at once

### Description of Task to be completed?

1. Pull the repo to get the project's latest version
2. Create a branch to work on
3. Select accommodation from the database.
4. Change` isBooked`  to be true and `userId` to be the id of a user who booked accommodation.
5. Write the whole CRUD
6. Validate User's Input
7. Write Tests
8. Add, commit, and push changes to GitHub

### How should this be manually tested?

1. Clone a repo, run `cd script-heroes-bn-backend` to go to the project's repo
2.  Run `npm install` to install all project dependencies then run `npm run dev` to run the project. 
3. Go to postman, then use the URL: `http://localhost:PORTNO/api/accomodations/book` . Make sure that you have `accomodationId` , `checkInDate`, `checkOutDate`, and `roomId` in the request's body.
4. Complete all `CRUD` operations

### Any background context you want to provide?

The request's body should look like this:
`{
 "accommodationId":2,
"roomId": 3,
    "checkInDate": "January 1, 2020",
    "checkOutDate": "Feb 2, 2020"
}`
### What are the relevant pivotal tracker stories?
[#174797543](https://www.pivotaltracker.com/n/projects/2465784)
